src/data_c.c4

```diff
@@ -439,6 +479,33 @@ SHMEM_PROF_DEF_CTX_PUT_N_SIGNAL_NBI(`mem')
     }                                                         \
   }
 
+#define SHMEM_DEF_IBPUT(STYPE,TYPE)                           \
+  void SHMEM_FUNCTION_ATTRIBUTES                              \
+  SHMEMX_FUNC_PROTOTYPE(STYPE##_ibput, TYPE *target,            \
+                       const TYPE *source, ptrdiff_t tst,     \
+                       ptrdiff_t sst, size_t bsize, size_t nblocks, int pe)  \
+    SHMEM_ERR_CHECK_INITIALIZED();                            \
+    SHMEM_ERR_CHECK_PE(pe);                                   \
+    SHMEM_ERR_CHECK_CTX(ctx);                                 \
+    SHMEM_ERR_CHECK_POSITIVE(tst);                            \
+    SHMEM_ERR_CHECK_POSITIVE(sst);                            \
+    SHMEM_ERR_CHECK_SYMMETRIC(target, sizeof(TYPE) * ((nblocks-1) * tst + 1)); \
+    SHMEM_ERR_CHECK_NULL(source, nblocks);                    \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(tst, bsize);             \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(sst, bsize);             \
+    SHMEM_ERR_CHECK_OVERLAP(target, source,                   \
+                   sizeof(TYPE) * ((nblocks-1) * tst + 1),     \
+                   sizeof(TYPE) * ((nblocks-1) * sst + 1), 0); \
+    size_t i = 0;                                             \
+    for ( ; nblocks > 0 ; --nblocks) {                        \
+      for (i = 0; i < bsize; i++) {                           \
+        shmem_internal_put_scalar(ctx, target++, source++,    \
+                                  sizeof(TYPE), pe);          \
```

Comment:
@philipmarshall21 - I understood this API to chunk puts into `bsize` segments, is that wrong?

Reply:
My understanding is that the API puts `nblocks` segments, each of size `bsize`. I think the **shmem_internal_put_scalar** can be replaced with something similar to what is **SHMEM_DEF_PUT** (except, `nelems` would be `bsize`; not sure if that is what you are hinting at):
![image](https://github.com/Sandia-OpenSHMEM/SOS/assets/97559536/60f17785-db8c-4326-af87-289a5d8d001b)


Although, this would require calling **shmem_internal_put_wait** `nblocks` times, which may be unavoidable.